### PR TITLE
feat: adding cli doc (generated from its yaml file) 50%

### LINKED ISSUE #78: feat: adding cli doc (generated from its yaml file) 50%
We want to generate a help page for the `pg` CLI tool (which used in tutorial). 

In another task (pinglue/pg-repo#27) we are storing the cli info a yaml file (`cmds.yaml`) to be a central source of truth for the cli info.

In this task we want to use this yaml file to generate a cli help page on the site.

we should create a sample page in `__mocks` folder. The main target here is a new component which you can call `CliDoc` which has a `data` prop which accepts a json equivalent to the data in cmds.yaml and renders a help doc for the cli. for your mock page, you can copy a json file equivalent of the cmd.yaml to the mock folder and import it in your page as its data.

Note that the current cmds.yaml is stub, you can add a few more commands to it from pg.ts for a better sample. There are online tools to convert yaml to json for this purpose.

format for cli page: there is a very standard format for cli help pages: first the general description followed by a list of commands. Then for each command there is a section that shows the details of that command with the same format: command title, description, list of options (flags + description). You can use [this](https://www.gatsbyjs.com/docs/reference/gatsby-cli/) to get some idea of how a cli help page looks like. TO give you more idea, the page is almost similar to running the following cmds (say in hello-world tutorial):

```
npx pg

# then for each command:
  npx pg help command
```

### PR DESCRIPTION
**Pull request by yuki-tylar**
Fixes #78

### CODE REVIEW

**Comment by yuki-tylar** [comment]
followed Gatsby's design.

* I added examples in yaml file.
* Currently, I'm using doc-page-res as mock

File: src/pages/__mocks/cli-doc.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	const headings = data.commands.map(item => { return {
+		depth: 2,
+		value: commandName(item.command)
+	}});
+
+	return (
+		<DocPageRes headings={headings}>
+			<h1>Commands (pinglue CLI)</h1>
```
Comment (pmvald) [inline src/pages/__mocks/cli-doc.tsx:38]: everything here should move to a component (within components folder) called `cli-doc` with a data prop (and type `Prop` as you defined here). Then you can import it in this mock page and display it.

File: src/pages/__mocks/cli-doc.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			command: string;
+		}[]
+		action?: string;
+	}[]
+}
+
+export default function (data: Props) {
+	data = cmdMock;
+
+	const commandName = (s: string) => {
```
Comment (pmvald) [inline src/pages/__mocks/cli-doc.tsx:27]: utility functions are better to be defined outside of the component function. This way they get compiled just once

File: src/pages/__mocks/cli-doc.tsx (hunk 1 above)
Comment (pmvald) [inline src/pages/__mocks/cli-doc.tsx:38]: We also need a section on the top for the main cli as well. Its format is the same as individual commands, but it info come from the top level fields in the yaml file: `description`, `command` (this last one I just added in a new commit). And then we show a table or list of the commands with the same format that is used for flags: we show the command usage (command field) of each command along with the command description. For command description, I think itis good we have a new field `shortDescription` that we use in this table. I added such a field for the first command. If a command does not have shortDescription then we fall back to its description. AFter this section we show section for individual fields starting with `<h2>CLI commands </h2>` (each command section begins with its name in `h3` as you already did)

File: src/pages/__mocks/cli-doc.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+						</h3>
+						<h4>Usage</h4>
+						<pre className="shell">
+							<code>
+								npx pg {item.command}
+							</code>
+						</pre>
+						<p>{item.description}</p>
+
+						<table className="table">
```
Comment (pmvald) [inline src/pages/__mocks/cli-doc.tsx:62]: The problem with table is it doesn't show well in mobile ... what does the Gatsby website do in mobile view? Maybe we use responsive columns here where the description falls under the flag in mobile view. See what is the best way

File: src/pages/__mocks/cli-doc.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+											<td>{option.flags}</td>
+											<td>{option.description}</td>
+										</tr>
+									))
+								}
+							</tbody>
+						</table>
+						{
+							item.examples ? <>
+								<h4>Examples</h4>
```
Comment (pmvald) [inline src/pages/__mocks/cli-doc.tsx:82]: +1 for examples

File: src/pages/__mocks/doc-page-res.tsx [hunk 5]
```diff
@@ -7,9 +7,10 @@ import classNames from 'classnames';
 import { Helmet } from 'react-helmet';
```
Comment (pmvald) [inline src/pages/__mocks/doc-page-res.tsx:7]: we can just create a new mock page to show cli doc. Why destroying this one? this mock could still be needed for testing more demos.

**Review by pmvald — CHANGES_REQUESTED** [review]
looks good. I made some comments

**Comment by yuki-tylar** [comment]
Updated.

* brought back `DocPageRes`
* copied `DocPageRes` into `CliDoc` in `__mocks` instead of modifying original `DocPageRes` . `CliDoc` in `__mocks` use `CliDocBody`
* created `CliDocBody` in `component` folder. This is contentBody in `<article class=s-content'></article>`
* created `CliDocItem`. this is reusable command section
* table is now responsive.

**Pull request merged**